### PR TITLE
Add workflow specification page

### PR DESCRIPTION
**Pull request by CLEckhardt**
Adds a "feature doc" for workflow, with content taken from the RFC.

Signed-off-by: Chris Eckhardt <eckhardt@bitwise.io>

### CODE REVIEW

File: _includes/0.3/left_sidebar.html [hunk 1]
```diff
@@ -58,6 +58,7 @@
         <a href="{% link docs/0.3/grid_product_smart_contract_specification.md %}">Product Specification</a>
```
Comment (shannynalayna) [inline _includes/0.3/left_sidebar.html:58]: There should also be a link to the feature document here https://grid.hyperledger.org/community/planning/roadmap.html

Comment (CLEckhardt) [reply _includes/0.3/left_sidebar.html:58]: Thanks, updated.

File: docs/0.3/workflow_specification.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    name: String,
+    constraints: Vec<String>,
+    permission_aliases: Vec<PermissionAlias>,
+    transitions: Vec<String>,
+}
+
+impl WorkflowState {
+    pub fn can_transition(&self, new_state: String,
+      pike_permissions: Vec<String>) -> bool {
+        if !self.transitions.contains(&new_state) {
```
Comment (shannynalayna) [inline docs/0.3/workflow_specification.md:104]: can you fix the formatting here to be : 
```
pub fn can_transition(
    &self,
    new_state: String,
    pike_permissions: Vec<String>,
) -> bool {...
```
to match what the rust compiler would fmt to (a little confusing to read otw)

Also, so we've added a check here for the workflow state's own name. Currently implemented in main. That change came after the RFC was merged, but I'm not sure if this document should stick to the RFC or current impl.

Comment (CLEckhardt) [reply docs/0.3/workflow_specification.md:107]: I've updated it to match the implementation and cargo formatting (which puts the args on one line)

**Review by shannynalayna — APPROVED** [review]

File: docs/0.3/workflow_specification.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+}
+```
+
+Below is an example of a permission alias with the name `po::seller` that has
+permission to perform two actions, update a purchase order and update its
+version number, and the ability to perform one transition, move the purchase
+order to it’s confirmed state:
+
+```rust
+let alias = PermissionAlias::new("po::seller");
```
Comment (shannynalayna) [inline docs/0.3/workflow_specification.md:177]: It looks like there is a `}` at the end of this code example that I don't think should be there

Comment (shannynalayna) [reply docs/0.3/workflow_specification.md:177]: Also, the permission name should match the thing being assigned in these lines below, so the following should be changed to `alias.add_permission(...)`. And, the indentation of these lines is not needed.

Comment (CLEckhardt) [reply docs/0.3/workflow_specification.md]: Thanks, updated this with a real permissions example.

**Review by dnewh — APPROVED** [review]

**Pull request merged**